src/main/java/org/radarcns/phone/PhoneUsageManager.java

```diff
@@ -0,0 +1,285 @@
+        // Create the pending intent and wrap our intent
+        PendingIntent pendingIntent = PendingIntent.getBroadcast(this.context, 1, intent, 0);
+
+        // Get alarm manager and schedule it to run every period (seconds)
+        AlarmManager alarmManager = (AlarmManager) this.context.getSystemService(ALARM_SERVICE);
+        alarmManager.setInexactRepeating(AlarmManager.RTC_WAKEUP, System.currentTimeMillis(), period * 1000, pendingIntent);
+
+        // Activity to perform when alarm is triggered
+        this.context.registerReceiver(new BroadcastReceiver() {
+            @Override
+            public void onReceive(Context context, Intent intent) {
+                processUsageEvents();
+            }
+        }, new IntentFilter("ACTIVITY_LAUNCH_WAKE"));
+
+        logger.info("Usage event alarm activated and set to a period of {} seconds", period);
+    }
+
+    private void processUsageEvents() {
+        // Get events from previous event to now or from a fixed history
+        final long queryStartTime = previousEventPackageName == null ? System.currentTimeMillis() - USAGE_EVENT_HISTORY_DEFAULT * 1000 : previousEventTimestamp;
+        final long queryEndTime = System.currentTimeMillis();
+        UsageEvents usageEvents = usageStatsManager.queryEvents(queryStartTime, queryEndTime);
+
+        // Loop through all events, send opening and closing of app
+        // Assume events are ordered on timestamp in ascending order (old to new)
+        UsageEvents.Event usageEvent = new UsageEvents.Event();
+        while (usageEvents.getNextEvent(usageEvent)) {
+            // Ignore config changes
+            if (usageEvent.getEventType() == UsageEvents.Event.CONFIGURATION_CHANGE) {
+                continue;
+            }
+
+            boolean isSent = false;
+            if (isNewUsageEvent(usageEvent)) {
+                // New event, so previous event was a closing of the previous event (?)
+                // Send this closing event
+                if (previousEventPackageName != null && !previousEventIsSent)
+                    sendUsageEvent(previousEventPackageName, previousEventTimestamp, previousEventType);
+
+                // Send the opening of new event
+                sendUsageEvent(usageEvent);
+                isSent = true;
+            }
+
+            // If not already processed, save
+            updatePreviousUsageEvent(usageEvent, isSent);
+
+            usageEvent = new UsageEvents.Event();
+        }
+
+        // If no event was found, then the permission needs to be given
+        if (previousEventPackageName == null){
+//            Intent intent = new Intent(Settings.ACTION_USAGE_ACCESS_SETTINGS);
+//            startActivity(intent);
+            logger.info("Please give this app Android Usage Access permissions (Security->App Usage access)");
+            Boast.makeText(context, "Please give this app Android Usage Access permissions (Security->App Usage access)", Toast.LENGTH_LONG).show();
+        }
```

Comment:
Should not be needed, use the provider `needsPermissions()` instead (see comment below).